docs/production/events.md

```diff
@@ -13,11 +13,114 @@ the following attributes:
+ status            | queue_status                | not null
+ dequeued_by       | integer                     |
+ canvas_id         | uuid                        | not null
+ account_id        | uuid                        | not null
+ space             | text                        | not null
+ name              | text                        | not null
+ value             | text                        | not null
+ retries           | integer                     | not null default 0
+ delay_until       | timestamp without time zone | not null default now()
+ modifier          | text                        | not null
+ enqueued_at       | timestamp with time zone    |
+ last_processed_at | timestamp with time zone    |
+```
+
+#### queue_status
+
+```
+enumlabel
+----------------------
+ new # When first added or retried.
+ scheduled # Ready to be picked up
+ locked # Does not appear to be used
+ done # Has succeeded
+ error # Has failed (won't be retried)
+ missing # Couldn't be found (won't be retried)
+```
+
+### Scheduling rules
+
+```
+ id           | integer                     | not null default nextval('scheduling_rules_id_seq'::regclass)
+ rule_type    | scheduling_rule_type        | not null
+ canvas_id    | uuid                        | not null
+ handler_name | text                        | not null
+ event_space  | text                        | not null
+ created_at   | timestamp without time zone | not null default now()
+```
+
+#### Scheduling_rules_type
+
+```
+enumlabel
+----------------------
+ pause
+ block
+```
+
+## Emit
+
+Done in `LibEvent` via `emit`, or automatically via `CronChecker`. Calls
+`EventQueue.enqueue`. This adds a new value to the events table with:
+
+- `status = 'new'`
+- `dequeued_by = null`
+- `delay_until = CURRENT_TIMESTAMP`
+- `enqueued_at = CURRENT_TIMESTAMP`
+
+Note that `CronChecker` does not use event table information to schedule, it uses
+`cron_records`.
+
+## Execution
+
+Fetch the request with by calling `Event.dequeue`. Searches for `status=scheduled`.
+
+When done, saved again with `status=done`.
+
+## Error handling
+
+Cases:
+- we tried to load the canvas and couldn't
+  - tracked in Telemetry with `event.load_success=false`
+  - if `retries < 2`
+    - increment `retries`
+    - put back with `status=new`
+  - else
+    - put back with `status=error`
+- error when running execute_handler
+  - tracked in Telemetry with `event.execution_success=false`
+  - same logic as loading the canvas
+- Incomplete:
+  - DOES NOT HAPPEN (removed at some point)
+  - set `status=new`
+  - set `delay_until` to 5 minutes from now.
+
+## Scheduling rules
+
+Code in `EventQueue.fs`. Allows a user to pause a queue, or allows an admin to lock a  queue for operational purposes.
+
+## Scheduling
+
+Code in `queue-scheduler/src/lib.rs`.
+
+Every second, finds `status=new` with `delay_until` reached, so long as there's no
+`block` or `pause` in this handler's `scheduling_rules`.
```

Comment:
This discusses the _finding_ but doesn't mention what happens once the finding is done

Reply:
Cool, will update. It sets the `status` to `scheduled`, which QueueWorker searches for.